src/OpenApi/perf/TransformersBenchmark.cs

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Diagnosers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.AspNetCore.OpenApi.Microbenchmarks;
+
+/// <summary>
+/// The following benchmarks are used to assess the memory and performance
+/// impact of different types of transformers. In particular, we want to
+/// measure the impact of (a) context-object creation and caching and (b)
+/// enumerator usage when processing operations in a given document.
+/// </summary>
+[MemoryDiagnoser]
+[EventPipeProfiler(EventPipeProfile.GcVerbose)]
+public class TransformersBenchmark : OpenApiDocumentServiceTestBase
+{
+    [Params(10, 100, 1000)]
+    public int InvocationCount { get; set; }
+
+    private readonly IEndpointRouteBuilder _builder = CreateBuilder();
+    private readonly OpenApiOptions _options = new OpenApiOptions();
+    private OpenApiDocumentService _documentService;
+
+    [GlobalSetup(Target = nameof(OperationTransformerAsDelegate))]
+    public void OperationTransformerAsDelegate_Setup()
+    {
+        _builder.MapGet("/", () => { });
+        for (var i = 0; i <= 1000; i++)
```

Comment:
I'm guessing the upper bound has to correspond to the maximum value of `InvocationCount`?  Why is it `<=`, rather than `<`?

Reply:
The parameterized value in these benchmarks is the number of times that the `GetOpenApiDocumentAsync` and transformers are invoked. We could parameterize the number of transformers that are registered as well, but I kept it fixed here.

Reply:
Right, but if we added `2000` to the list above, we'd have to change this, right?

Reply:
Not necessarily -- since the transformer count is fixed in all these tests. There's two dimensions to the test (a) how many transformers are registered and (b) how many times they entire transformer pipeline is invoked. This particular setup validates the later.